pkg/cache/context.go

```diff
@@ -295,16 +295,25 @@ func (ctx *Context) triggerReloadConfig() {
 	}
 }
 
-func (ctx *Context) updatePodCondition(pod *v1.Pod, condition *v1.PodCondition) error {
-	log.Logger.Info("Updating pod condition",
+// update pod condition if the condition has changed, it returns a bool value to indicate
+// if the update is made or skipped, return an error if it is unable to make the change when it is needed to.
+func (ctx *Context) updatePodCondition(pod *v1.Pod, condition *v1.PodCondition) (bool, error) {
```

Comment:
This function should be merged into the caller, or the caller should be merged into this one. There are two layers of functions, two checks for errors and there is a 1:1 relation between `updatePodConditionFn` and `updatePodCondition`.
Too complex, does not make sense and uses a number of extra lines.

Reply:
Make sense to me, I've removed this function and merged it to the caller.